src/dune_rules/lib.ml

```diff
@@ -775,7 +775,7 @@ module rec Resolve_names : sig
   val resolve_dep :
     db -> Loc.t * Lib_name.t -> private_deps:private_deps -> lib Resolve.Memo.t
 
-  val resolve_name : db -> Lib_name.t -> Status.t Memo.t
+  val resolve_name : db -> Lib_name.t -> (Status.t * db) Memo.t
```

Comment:
Can you explain why we need this change?

Reply:
This is the last `db` where the `lib_name` where searched(or resolved) and with that, we can determine if the `lib_name` is external or not(external if`db.parent` is None).

Yeah I see your concern. I think it would it be better to expand Status.t by having `External` in it and remove `db`.